### PR TITLE
[Merged by Bors] - style(Algebra): write `coe_pow` lemmas more clearly

### PR DESCRIPTION
**Pull request by eric-wieser**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by fpvandoorn** [comment]
I personally don't think these fixes look better, but I don't mind them if they are necessary if we want to get a temporary power-parser fix in.

File: Mathlib/Algebra/Order/Interval.lean [hunk 1]
```diff
@@ -304,7 +304,7 @@ namespace NonemptyInterval
 
 @[to_additive]
 theorem coe_pow_interval [OrderedCommMonoid α] (s : NonemptyInterval α) (n : ℕ) :
-    (s ^ n : Interval α) = (s : Interval α) ^ n :=
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Interval.lean:307]: @fpvandoorn: my opinion is that this version is very confusing, it reads "somewhere within these parentheses there is a coercion, I hope you can guess where". In order to know where the coercion ends up, you have to know the details of the `^` elaborator. The new version does not have this property.

Comment (fpvandoorn) [reply Mathlib/Algebra/Order/Interval.lean:307]: That makes sense. I personally read the (_ : _) themselves as coercions (and place them on the smallest subterm that has that type), but it is indeed not guaranteed that the coercion is there.

Comment (eric-wieser) [reply Mathlib/Algebra/Order/Interval.lean:307]: Unfortunately I think your heuristic is bad, and you should stop using it:
```lean
import Mathlib.Algebra.Group.Defs
import Mathlib.Data.PNat.Basic

def add3 {α} [Add α] (a b c : α) : α := a + b + c

-- this is a tautology, the coercion is on the inside on both sides
example (a b c : PNat) : (add3 a b c : Nat) = add3 (a : Nat) b c := sorry
```

Comment (fpvandoorn) [reply Mathlib/Algebra/Order/Interval.lean:307]: That is fair, and I would indeed not write that LHS. The double type ascription is a possible alternative, but it is very verbose...
```
example (a b c : PNat) : ((add3 a b c : PNat) : Nat) = add3 (a : Nat) b c := sorry
```
`↑` can also be misleading, it doesn't necessarily insert a coercion there. But it's probably more reliable than type ascriptions.

bors d+

**Comment by kmill** [comment]
Please make sure this is synchronized with the efforts in #8284 and won't make the Lean version bump any more difficult.

**Comment by eric-wieser** [comment]
I will synchronize https://github.com/leanprover-community/mathlib4/pull/8284 with this once it lands (hopefully before you wake up!). I expect the merge conflicts to be easy.

bors r+

**Comment by kmill** [comment]
Eric, I would really prefer that you *first* look at the fixes PR and then transfer changes to this one. I am against merging this right now.

**Comment by eric-wieser** [comment]
As predicted this didn't contain any conflicts with the core bump; it's now largely vacuous; can you either reject or merge the lingering style changes, @kmill?

**Comment by kmill** [comment]
Thanks for waiting until the Lean version bump went through. On the flip side to your prediction that it would not have conflicts, that means it wasn't necessary trying to get it in a week or so sooner, right? But, since you brought it up, I feel that I need to point out that it's factually incorrect that there are no conflicts here. I tested merging master from e2885eb, and there are merge conflicts with the "chore: bump to v4.3.0-rc2" commit, exactly as I feared.

I've verified that these lingering changes yield the same types, and they seem fine to me.

bors r+

**Pull request closed without merging**